jax/lax_linalg.py

```diff
@@ -179,28 +178,53 @@ def eig_abstract_eval(operand):
     w = ShapedArray(batch_dims + (n,), dtype)
   else:
     raise NotImplementedError
-  return w, vl, vr
+
+  if jobvl and jobvr:
```

Comment:
I wonder if we could write this more succinctly? I think the list append idiom below is a bit shorter.

Reply:
On George's advice, I'm only dealing with `lapack.pyx` in this PR. The other points will be addressed in a further PR.